server/auth_server.go

```diff
@@ -365,6 +552,79 @@ func UpgradeSettings() error {
 	return nil
 }
 
+func UpgradeCase() error {
+	var settings []string
```

Comment:
Would it make more sense to do the 
```
authColl, err := RancherClient.GenericObject.List(&client.ListOpts{
 +			Filters: filters,
 +		})
 +		if err != nil {
 +			log.Errorf("Error getting the go 'auth.config', error: %v", err)
 +			return err
 +		}
 +
 +		if len(authColl.Data) > 0 {
 +			log.Info("Config stored")
 +			return nil
 +		}
```
logic right up front at the beginning of the function so that you dont do extra processing after its saved for the first time?